apps/api/v2/src/modules/billing/controllers/billing.controller.ts

```diff
@@ -0,0 +1,109 @@
+import { AppConfig } from "@/config/type";
+import { BaseApiResponseDto } from "@/lib/response/response.dto";
+import { Roles } from "@/modules/auth/decorators/roles/roles.decorator";
+import { NextAuthGuard } from "@/modules/auth/guards/next-auth/next-auth.guard";
+import { OrganizationRolesGuard } from "@/modules/auth/guards/organization-roles/organization-roles.guard";
+import { BillingService } from "@/modules/billing/billing.service";
+import { SubscribeToPlanInput } from "@/modules/billing/controllers/inputs/subscribe-to-plan.input";
+import { CheckPlatformBillingResponseDto } from "@/modules/billing/controllers/responses/CheckPlatformBillingResponse.dto";
+import { SubscribeTeamToBillingResponseDto } from "@/modules/billing/controllers/responses/SubscribeTeamToBillingResponse.dto";
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Get,
+  Param,
+  Post,
+  Req,
+  UseGuards,
+  Headers,
+  HttpCode,
+  HttpStatus,
+} from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { Request } from "express";
+import { Stripe } from "stripe";
+
+import { ApiResponse } from "@calcom/platform-types";
+
+@Controller({
+  path: "/billing",
+  version: "2",
+})
+export class BillingController {
+  private readonly stripeWhSecret: string;
+
+  constructor(
+    private readonly billingService: BillingService,
+    private readonly configService: ConfigService<AppConfig>
+  ) {
+    this.stripeWhSecret = configService.get("stripe.webhookSecret", { infer: true }) ?? "";
+  }
+
+  @Get("/:teamId/check")
+  @UseGuards(NextAuthGuard, OrganizationRolesGuard)
+  @Roles(["OWNER", "ADMIN"])
+  async checkTeamBilling(@Param("teamId") teamId: number): Promise<CheckPlatformBillingResponseDto> {
+    const teamBilling = await this.billingService.getBillingData(teamId);
+
+    return {
+      status: "success",
+      valid: teamBilling.status === "valid",
+    };
+  }
+
+  @Post("/:teamId/subscribe")
+  @UseGuards(NextAuthGuard, OrganizationRolesGuard)
+  @Roles(["OWNER", "ADMIN"])
+  async subscribeTeamToStripe(
+    @Param("teamId") teamId: number,
+    @Body() input: SubscribeToPlanInput
+  ): Promise<SubscribeTeamToBillingResponseDto> {
```

Comment:
should we use ApiResponse type here

Reply:
Promise<ApiResponse<SubscribeTeamToBillingResponseDto>>

Reply:
Doesn't look like we're using ApiResponse much anymore, so I don't see a point on using it when we're already shaping responses as outputs as classes.

Reply:
the main goal is to force a standard on the shape of api responses
 and so that that the DTO only targets the actual data sent back

Promise<ApiResponse < DTO> >

Reply:
apiResponse could be used directly in the dto as well I think, as you see fit